### ISSUE TITLE
padding_side='left' in collator

### ISSUE THREAD

**Issue by shreydan** [huggingface/nanoVLM#29]
in VQACollator the padding_side is 'left':

```py
encoded_full_sequences = self.tokenizer.batch_encode_plus(
    input_sequences,
    padding="max_length",
    padding_side="left",
    return_tensors="pt",
    truncation=True,
    max_length=self.max_length,
)
```
so now in model.forward(), the image embeddings and text embeddings are being concatenated
```py
combined_embd = torch.cat((image_embd, token_embd), dim=1)
```
so for sequences in the batch which have left-padding, the pad tokens will be between the image tokens and the text tokens, which may not be efficient/suitable for causal-attention. 

This can be fixed with padding_side='right' or by using placeholder `<image>` tokens added to text.

Thank you.

**Comment by andimarafioti** [comment]
Good catch! This is definitely something we have to fix! The repo also doesn't have proper packing. But this would be a good step in the right direction. We are already discussing with @lusxvr how to do it, but if someone from the community wants to take this small fix, we'd be happy :)

**Comment by lusxvr** [comment]
Agreed! The way to go in my opinion is using placeholder tokens for the images.
